src/aiida/tools/dumping/processes.py

```diff
@@ -0,0 +1,448 @@
+            )
+
+        else:
+            process_dumped = False
+
+        all_processes_dumped = all_processes_dumped and process_dumped
+
+    return all_processes_dumped
+
+
+# Separate functions for CalcJob dumping using pre_submit, as well as for the node_inputs
+def calcjob_node_inputs_dump(calcjob_node: CalcJobNode, output_path: Path, inputs_relpath: Path = Path('node_inputs')):
+    """Dump inputs of a `CalcJobNode` of type `SinglefileData` and `FolderData`.
+
+    :param calcjob_node: The `CalcJobNode` whose inputs will be dumped.
+    :type calcjob_node: CalcJobNode
+    :param output_path: The path where the inputs will be dumped.
+    :type output_path: Path
+    """
+    dump_types = (SinglefileData, FolderData)
+
+    # ? Not using the `node_class` argument of `get_incoming`, as it does not actually retrieve, e.g. a `UpfData` node
+    # ? (due to planned deprecation?)
+    # ? Instead, check for isinstance of `SinglefileData`
+    input_node_triples = calcjob_node.base.links.get_incoming(link_type=LinkType.INPUT_CALC)
+
+    for input_node_triple in input_node_triples:
+        # Select only repositories that hold objects and are of the selected dump_types
+        if len(input_node_triple.node.base.repository.list_objects()) > 0 and isinstance(
+            input_node_triple.node, dump_types
+        ):
+            # input_node_path = output_path / Path('node_inputs') / Path(input_node_triple.link_label)
+            input_node_path = output_path / inputs_relpath / Path(*input_node_triple.link_label.split('__'))
+            input_node_triple.node.base.repository.copy_tree(input_node_path)
+
+
+def calcjob_presubmit_dump(calcjob_node: CalcJobNode, output_path: Path):
+    """
+    Dump inputs of a `CalcJobNode` using the `presubmit` function.
+
+    :param process: The `CalcJobNode` whose inputs need to be dumped.
+    :type process: CalcJobNode
+    :param output_path: The path where the inputs will be dumped.
+    :type output_path: Path
+    """
+
+    builder_restart = calcjob_node.get_builder_restart()
+    runner = get_manager().get_runner()
+    calcjob_process: CalcJob = instantiate_process(runner, builder_restart)  # type: ignore[assignment]
+
+    # `presubmit` calls `prepare_for_submission` internally
+    calc_info = calcjob_process.presubmit(folder=Folder(abspath=output_path))
+
+    try:
+        # Hackish way to modify local copy list so that the pseudos are actually dumped where I want them to. Otherwise
+        # they
+        # end up in home...
+        local_copy_list = calc_info['local_copy_list'].copy()
+        # print('LOCAL_COPY_LIST', local_copy_list)
+        new_local_copy_list = [tuple(list(local_copy_list[0][:2]) + [str(output_path / local_copy_list[0][-1])])]
+        calc_info['local_copy_list'] = new_local_copy_list
+        # print('NEW_LOCAL_COPY_LIST', new_local_copy_list)
+    except IndexError:
+        # This happens if `local_copy_list` is empty
+        pass
```

Comment:
This is indeed very hackish and is bound to fail for examples that you didn't consider. The problem is that the target in the `local_copy_list` instructions is typically _relative_. Furthermore, you are relying on an implementation detail of `upload_calculation` that the files specifiec in `local_copy_list` are written to the `folder` sandbox. This _used_ to be the case, but was actually recently changed, so I think that if you rebase this branch on the latest `main`, your local copy list files won't even be written to the `output_path` anymore.

The `local_copy_list` is now copied directly to the working directory of the calculation using the transport. This working directory is automatically determined on the UUID of the calculation and the base path returned by the `get_workdir()` method of the computer on which the calculation is running. See this code: https://github.com/aiidateam/aiida-core/blob/main/src/aiida/engine/daemon/execmanager.py#L109-L170

When you mentioned to me that you wanted to actually run `upload_calculation` I already warned that it would most likely have to be refactored in order for it to work. I was thinking exactly of this part of the code. I was surprised that you didn't really have to, but that turns out to be merely because the implementation had an incorrect implementation, which very coincidentally I just had to correct very recently. We would have to factor out the determination of the working directory and changing the transport's current working directory to it, so it can be replaced/overridden by your dump method.

Reply:
Ah, that explains why it recently broke. Before, I didn't need this hackish code. Then I rebased, and it stopped working, and I added this abomination as a quick fix in the morning of the coding days to allow for some dogfooding :D So how should we proceed here now? Just remove the option for now, keep it in mind, and add it at a later stage?

Reply:
It would definitely make it easier to get the rest of this feature in the release if we but the `--presubmit` option on the backburner for now until we can find a proper solution. So I would opt for that route I think

Reply:
That's fine for me 👍